src/mono/msbuild/apple/build/AppleBuild.targets

```diff
@@ -110,6 +88,45 @@
     </ItemGroup>
   </Target>
 
+  <Target Name="_ScanAssembliesDecideLightweightMarshaler">
+    <ItemGroup>
+      <AssembliesToScan Include="@(_AssembliesToBundleInternal)" />
+    </ItemGroup>
+
+    <MarshalingPInvokeScanner Assemblies ="@(AssembliesToScan)">
+      <Output TaskParameter="IncompatibleAssemblies" ItemName="MonoLightweightMarshallerIncompatibleAssemblies" />
+    </MarshalingPInvokeScanner>
+    
+    <ItemGroup Condition="'@(MonoLightweightMarshallerIncompatibleAssemblies->Count())' > 0">
+      <RuntimeComponents Include="marshal-ilgen" />
+    </ItemGroup>
+    <ItemGroup Condition="'@(MonoLightweightMarshallerIncompatibleAssemblies->Count())' == 0">
+      <RuntimeComponents Remove="marshal-ilgen" />
+    </ItemGroup>
```

Comment:
Is there an assumption here that `RuntimeComponents` is non-empty and can already include `marshal-ilgen` component before reaching this target?
If yes, wouldn't we then encounter a duplication in case `<RuntimeComponents Include="marshal-ilgen" />` on line 101?

Reply:
Oh, `Include` does not check for duplicates?

Reply:
Unfortunately... no :)

Reply:
Apparently one can use `KeepDuplicates` on ItemGroups. Who knew